src/pip/_internal/operations/install/wheel.py

```diff
@@ -75,6 +75,8 @@ def save(self) -> None:
 RecordPath = NewType("RecordPath", str)
 InstalledCSVRow = Tuple[RecordPath, str, Union[int, str]]
 
+SHEBANG_PYTHON = re.compile(b"^#!python[w]*(\s.*)?$")
```

Comment:
`[w]*` seems wrong here. Surely `?` would be better, we don't want to pick up `pythonwwwwwww`. There's also things like `python3`, etc.

I'd argue that we should retain `startswith("#!python"` and then just do something like

```
parts = firstline.split(maxsplit=1)
postinterp = parts[1] if len(parts) > 1 else b""
```

The [spec](https://packaging.python.org/specifications/binary-distribution-format/#recommended-installer-features) says

> The b'#!pythonw' convention is allowed. b'#!pythonw' indicates a GUI script instead of a console script.

But we've not done that before now, so I'm inclined not to start. It's messy (not impossible, but messy) to determine if there even is a GUI version of the current Python interpreter in a cross-platform way, and no-one has ever complained, so I'd say we wait for someone with a use case to speak up.

I'd argue that the spec needs tightening up *a lot*, but I suspect no-one cares enough, so let's just do the minimum needed here.

Reply:
Ok. This sounds fine. I'll make the changes.